src/utils/walletConnect.js

```diff
@@ -0,0 +1,295 @@
+  }
+  const { peerId, peerMeta, chainId } = payload.params[0];
+
+  const dappName = peerMeta?.name;
+  const dappUrl = peerMeta?.url;
+  const dappScheme = peerMeta?.scheme;
+
+  const meta = {
+    chainId,
+    dappName,
+    dappScheme,
+    dappUrl,
+    peerId,
+  };
+
+  dispatch(setSession({ uri, sessionInfo: { meta } }));
+
+  Navigation.handleAction(Routes.WALLET_CONNECT_APPROVAL_SHEET, {
+    ...routeParams,
+    meta,
+  });
+};
+
+export const callRequestHandlerFactory = (dispatch, getState) => {
+  const modules = [
+    ...TransactionModule(dispatch, getState),
+    ...InformationModule(dispatch, getState),
+    ...ConnectionModule(dispatch, getState),
+  ];
+  const walletConnectHandlers = modules.reduce(
+    (acum, handlerObj) => ({
+      ...acum,
+      [handlerObj.methodName]: [handlerObj.handler, handlerObj.executor],
+    }),
+    {}
+  );
+
+  return methodName => {
+    const handler = walletConnectHandlers[methodName];
+    if (!handler) {
+      return [];
+    }
+
+    return handler;
+  };
+};
+
+export const initializeProtectedIds = async () => {
+  const nftCanisters = await getDabNfts();
+  const tokenCanisters = await getDabTokens();
+  const PROTECTED_IDS = [
+    ...(nftCanisters || []).map(collection =>
+      collection.principal_id.toString()
+    ),
+    ...(tokenCanisters || []).map(token => token.principal_id.toString()),
+    ...ASSET_CANISTER_IDS,
+  ];
+  await setProtectedIds(PROTECTED_IDS);
+};
+
+export const getDabTokens = async () => {
+  const agent = new HttpAgent({ fetch, host: IC_URL_HOST });
+  const tokens = await getTokens({ agent });
+  const parsedTokens = (tokens || []).map(token =>
+    recursiveParseBigint(recursiveParsePrincipal(token))
+  );
+  return parsedTokens.map(token => ({
+    ...token,
+    canisterId: token?.principal_id,
+  }));
+};
+
+export const getDabNfts = async () => {
+  const agent = new HttpAgent({ fetch, host: IC_URL_HOST });
+  return getAllNFTS({ agent });
+};
+
+export const fetchCanistersInfo = async whitelist => {
+  if (whitelist && whitelist.length > 0) {
+    const canistersInfo = await Promise.all(
+      whitelist.map(async id => {
+        let canisterInfo = { id };
+
+        try {
+          const fetchedCanisterInfo = await PlugController.getCanisterInfo({
+            canisterId: id,
+            fetch,
+          });
+          canisterInfo = { id, ...fetchedCanisterInfo };
+        } catch (error) {
+          console.error(error);
+        }
+
+        return canisterInfo;
+      })
+    );
+
+    const sortedCanistersInfo = canistersInfo.sort((a, b) => {
+      if (a.name && !b.name) {
+        return -1;
+      }
+      return 1;
+    });
+
+    return sortedCanistersInfo;
+  }
+
+  return [];
+};
+
+export const validateTransferArgs = ({ to, amount, opts, strAmount }) => {
+  let message = null;
+
+  if (amount && !validateAmount(amount)) {
+    message =
+      'The transaction failed because the amount entered was invalid. \n';
```

Comment:
we should move all texts to the i18n file. also, we shouldn't use '\n' to add spaces in screen, it'll bring issues in some screens

Reply:
They are internal error msg, we doesnt need to translate it